pkg/app/app.go

```diff
@@ -1352,6 +1357,14 @@ Do you really want to apply?
 	if !interactive || interactive && r.askForConfirmation(confMsg) {
 		r.helm.SetExtraArgs(argparser.GetArgs(c.Args(), r.state)...)
 
+		for _, release := range releasesWithPreApply {
+			a.Logger.Infof("\nRunning preapply hook for %s:", release.Name)
+			if _, err := st.TriggerPreapplyEvent(&release, "apply"); err != nil {
+				applyErrs = append(applyErrs, err)
+				continue
+			}
+		}
```

Comment:
To be extra correct, I think you should trigger preapply events in the order of processing, and only on releases that are actually processed.
That said, would you mind trying to remove this block, along with the `getReleasesWithPreApply` function, and instead add:

```
		for _, release := range rs{
			a.Logger.Infof("\nRunning preapply hook for %s:", release.Name)
			if _, err := st.TriggerPreapplyEvent(&release, "apply"); err != nil {
				applyErrs = append(applyErrs, err)
				continue
			}
		}
```

to L1379 and L1401?

Reply:
@Sajfer Well, it's almost a half year since we started reviewing this PR! Sorry about that.
If you want me to make the above change, test the outcome, and decide if my suggested change works, please let me know. I'll do my best.